package/MDAnalysis/analysis/base.py

```diff
@@ -220,7 +224,176 @@ def __setstate__(self, state):
+        ----------
+        start : int, optional
+            start frame of analysis
+        stop : int, optional
+            stop frame of analysis
+        step : int, optional
+            number of frames to skip between each analysed frame
+        frames : array_like, optional
+            array of integers or booleans to slice trajectory; `frames` can
+            only be used *instead* of `start`, `stop`, and `step`. Setting
+            *both* `frames` and at least one of `start`, `stop`, `step` to a
+            non-default value will raise a :exc:`ValueError`.
+        verbose : bool, optional
+            Turn on verbosity
+        reset_timestep : bool, optional
+            Reset the timestep object after for each ``analysis_object``.
+            Setting this to ``False`` can be useful if an ``analysis_object``
+            is performing a trajectory manipulation which is also useful for the
+            subsequent ``analysis_instances`` e.g. unwrapping of molecules.
+        """
+
+        # Ensure compatibility with API of version 0.15.0
+        if not hasattr(self, "_analysis_instances"):
+            self._analysis_instances = (self,)
+
+        logger.info("Choosing frames to analyze")
+        # if verbose unchanged, use class default
+        verbose = getattr(self, "_verbose", False) if verbose is None else verbose
+
+        logger.info("Starting preparation")
+
+        for analysis_object in self._analysis_instances:
+            analysis_object._setup_frames(
+                analysis_object._trajectory,
+                start=start,
+                stop=stop,
+                step=step,
+                frames=frames,
+            )
+            analysis_object._prepare()
+
+        logger.info(
+            f"Starting analysis loop over {self._analysis_instances[0].n_frames} "
+            "trajectory frames."
+        )
+
+        for i, ts in enumerate(
+            ProgressBar(self._analysis_instances[0]._sliced_trajectory, verbose=verbose)
+        ):
+            if reset_timestep:
+                ts_original = ts.copy()
+
+            for analysis_object in self._analysis_instances:
+                # Set attributes before calling `_single_frame()`. Setting
+                # these attributes explicitly is mandatory so that each
+                # instance can access the information of the current timestep.
+                analysis_object._frame_index = i
+                analysis_object._ts = ts
+                analysis_object.frames[i] = ts.frame
+                analysis_object.times[i] = ts.time
+
+                # Call the actual analysis of each instance.
+                analysis_object._single_frame()
+
+                if reset_timestep:
+                    ts = ts_original
+
+        logger.info("Finishing up")
+
+        for analysis_object in self._analysis_instances:
+            analysis_object._conclude()
+
+        return self
+
+
+class AnalysisBase(AnalysisCollection):
```

Comment:
I agree with @yuxuanzhuang : it feels very weird and circular to have AnalysisBase inherited from AnalysisCollection. It just does not make sense semantically.

Can this be changed?

Reply:
The only thing why I did this to do not copy code of the `run` method between `AnalysisBase` and `AnalysisCollection`. We can remove the inheritance by explicitly leaving the run method in `AnalysisBase` untouched. This would be fine for me!